packagedb/management/commands/fix_purl_values.py

```diff
@@ -0,0 +1,226 @@
+            name=name,
+            version=version,
+            qualifiers=qualifiers
+        )
+        urls = get_urls(
+            namespace=namespace,
+            name=name,
+            version=version,
+            qualifiers=self.package_url.qualifiers,
+        )
+        self.download_url = urls['repository_download_url']
+        self.repository_homepage_url = urls['repository_homepage_url']
+        self.api_data_url = urls['api_data_url']
+
+        qualifiers_mapping = self.package_url.qualifiers
+        filename = build_filename(
+            artifact_id=name,
+            version=version,
+            extension=qualifiers_mapping.get('type') or 'jar',
+            classifier=qualifiers_mapping.get('classifier'),
+        )
+        timestamps_by_links = get_timestamps_by_links(self.repository_homepage_url)
+        self.release_date = timestamps_by_links.get(filename)
+        self.related_artifacts = list(
+            self._populate_related_artifacts(
+                namespace=namespace,
+                name=name,
+                version=version,
+                ec=ec,
+            )
+        )
+
+    @classmethod
+    def _populate_related_artifacts(cls, namespace, name, version, ec):
+        filtered_ec = [entry for entry in ec if not entry.startswith('.')]
+        for entry in filtered_ec:
+            _, ending = entry.split('-')
+            split_ending = ending.split('.')
+            classifier = None
+            if len(split_ending) > 0:
+                classifier = split_ending[0]
+                qualifiers = f'classifier={classifier}'
+                yield cls(
+                    namespace=namespace,
+                    name=name,
+                    version=version,
+                    qualifiers=qualifiers,
+                )
+
+
+# This is from https://stackoverflow.com/questions/4856882/limiting-memory-use-in-a-large-django-queryset/5188179#5188179
+class MemorySavingQuerysetIterator(object):
+    def __init__(self,queryset,max_obj_num=1000):
+        self._base_queryset = queryset
+        self._generator = self._setup()
+        self.max_obj_num = max_obj_num
+
+    def _setup(self):
+        for i in range(0,self._base_queryset.count(),self.max_obj_num):
+            # By making a copy of of the queryset and using that to actually access
+            # the objects we ensure that there are only `max_obj_num` objects in
+            # memory at any given time
+            smaller_queryset = copy.deepcopy(self._base_queryset)[i:i+self.max_obj_num]
+            logger.debug('Grabbing next %s objects from DB' % self.max_obj_num)
+            for obj in smaller_queryset.iterator():
+                yield obj
+
+    def __iter__(self):
+        return self._generator
+
+    def next(self):
+        return self._generator.next()
+
+
+def query_sha1_on_maven(sha1, timeout=DEFAULT_TIMEOUT):
+    maven_api_search_url = f'https://search.maven.org/solrsearch/select?q=1:{sha1}'
```

Comment:
We have an issue here on this: https://github.com/nexB/purldb/issues/498 what do you think about this?